File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
import java.util.*;
class Solution {
    public int[] sortedSquares(int[] nums) {

        List<Integer> pos = new ArrayList<>();
        List<Integer> neg = new ArrayList<>();

        for(int num : nums)
        {
            if(num >= 0)
            {
                pos.add(num*num);
            }
            else
            {
                neg.add(num*num);
            }
        }

        Collections.reverse(neg);

        int i = 0;
        int j = 0;
        int idx= 0;
        int n = neg.size();
        int m = pos.size();
        int[] res = new int[nums.length];

        while(i < m && j < n)
        {
            if(pos.get(i) <=  neg.get(j))
            {
                res[idx] = pos.get(i);
                idx++;
                i++;
            }
            else
            {
                res[idx] = neg.get(j);
                idx++;
                j++;
            }
        }
        
        while(i < m)
        {
            res[idx] = pos.get(i);
            idx++;
            i++;
        }
        while(j < n)
        {
            res[idx] = neg.get(j);
            idx++;
            j++;
        }

        return res;
    }
}

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:

        pos =[]
        neg =[]

        for num in nums:
            if num >=0:
                pos.append(num)
            else:
                neg.append(num)

        for i in range(len(pos)):
            pos[i] = pos[i]*pos[i]
        for j in range(len(neg)):
            neg[j] = neg[j]*neg[j]

        neg.reverse()

        i=0
        j=0
        res = [] 
        m = len(pos)
        n = len(neg)

        while i<m and j<n:

            if pos[i] <= neg[j]:
                res.append(pos[i])
                i+=1
            else:
                res.append(neg[j])
                j+=1

        while i<m:
            res.append(pos[i])
            i+=1

        while j<n:
            res.append(neg[j])
            j+=1
                    
        return res            

                   
        